drivers/sensor/iis2iclx/iis2iclx.c

```diff
@@ -0,0 +1,657 @@
+/* ST Microelectronics IIS2ICLX 6-axis IMU sensor driver
+ *
+ * Copyright (c) 2020 STMicroelectronics
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * Datasheet:
+ * https://www.st.com/resource/en/datasheet/iis2iclx.pdf
+ */
+
+#define DT_DRV_COMPAT st_iis2iclx
+
+#include <drivers/sensor.h>
+#include <kernel.h>
+#include <device.h>
+#include <init.h>
+#include <string.h>
+#include <sys/byteorder.h>
+#include <sys/__assert.h>
+#include <logging/log.h>
+
+#include "iis2iclx.h"
+
+LOG_MODULE_REGISTER(IIS2ICLX, CONFIG_SENSOR_LOG_LEVEL);
+
+static const uint16_t iis2iclx_odr_map[] = {0, 12, 26, 52, 104, 208, 416, 833,
+					1660, 3330, 6660};
+
+#if defined(IIS2ICLX_ACCEL_ODR_RUNTIME)
+static int iis2iclx_freq_to_odr_val(uint16_t freq)
+{
+	size_t i;
+
+	for (i = 0; i < ARRAY_SIZE(iis2iclx_odr_map); i++) {
+		if (freq == iis2iclx_odr_map[i]) {
+			return i;
+		}
+	}
+
+	return -EINVAL;
+}
+#endif
+
+static int iis2iclx_odr_to_freq_val(uint16_t odr)
+{
+	/* for valid index, return value from map */
+	if (odr < ARRAY_SIZE(iis2iclx_odr_map)) {
+		return iis2iclx_odr_map[odr];
+	}
+
+	/* invalid index, return last entry */
+	return iis2iclx_odr_map[ARRAY_SIZE(iis2iclx_odr_map) - 1];
+}
+
+#ifdef IIS2ICLX_ACCEL_FS_RUNTIME
+static const uint16_t iis2iclx_accel_fs_map[] = {500, 3000, 1000, 2000};
+static const uint16_t iis2iclx_accel_fs_sens[] = {1, 8, 2, 4};
+
+static int iis2iclx_accel_range_to_fs_val(int32_t range)
+{
+	size_t i;
+
+	for (i = 0; i < ARRAY_SIZE(iis2iclx_accel_fs_map); i++) {
+		if (range == iis2iclx_accel_fs_map[i]) {
+			return i;
+		}
+	}
+
+	return -EINVAL;
+}
+#endif
+
+static inline int iis2iclx_reboot(const struct device *dev)
+{
+	struct iis2iclx_data *data = dev->data;
+
+	if (iis2iclx_boot_set(data->ctx, 1) < 0) {
+		return -EIO;
+	}
+
+	/* Wait sensor turn-on time as per datasheet */
+	k_busy_wait(35 * USEC_PER_MSEC);
```

Comment:
Do you need to busy wait or can you use sleep instead?

Reply:
I don't need to busy wait. I'll use k_sleep instead.